framework/wazuh/core/batcher/batcher.py

```diff
@@ -0,0 +1,196 @@
+        except Exception as e:
+            tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
+            logger.error(f"Error when sending buffer:\n{''.join(tb_str)}")
+
+    def create_flush_buffer_task(self):
+        """
+        Creates a task to flush the current buffer and reset it. This task sends the buffered messages to the indexer
+        and clears the buffer.
+        """
+        buffer_copy = self._buffer.copy()
+        asyncio.create_task(self._send_buffer(buffer_copy))
+        self._buffer.reset()
+
+    async def run(self):
+        """
+        Continuously retrieves messages from the queue and batches them based on the configuration. Handles signals
+        for shutdown and manages the batching logic.
+
+        Handles:
+        - Retrieving messages from the queue.
+        - Checking buffer limits.
+        - Creating and managing flush tasks.
+        - Handling shutdown signals.
+        """
+        # Register signal handlers
+        self._shutdown_event = asyncio.Event()
+        loop = asyncio.get_running_loop()
+        loop.add_signal_handler(signal.SIGINT, self._handle_signal, signal.SIGINT)
+        loop.add_signal_handler(signal.SIGTERM, self._handle_signal, signal.SIGTERM)
+
+        try:
+            while not self._shutdown_event.is_set():
+                done, pending = await asyncio.wait(
+                    [self._get_from_queue(), self._timer.wait_timeout_event()],
+                    return_when=asyncio.FIRST_COMPLETED
+                )
+
+                # Process completed tasks
+                for task in done:
+                    if isinstance(task.result(), Message):
+                        message = task.result()
+
+                        if self._buffer.get_length() == 0:
+                            self._timer.create_timer_task()
+
+                        self._buffer.add_message(message)
+
+                        if self._buffer.check_count_limit() or self._buffer.check_size_limit():
+                            self.create_flush_buffer_task()
+                            self._timer.reset_timer()
+                    else:
+                        # Cancel the reading task if it is still pending
+                        for p_task in pending:
+                            p_task.cancel()
+
+                        self.create_flush_buffer_task()
+                        self._timer.reset_timer()
+
+        except Exception as e:
+            tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
+            logger.error(f"Error in batcher loop:\n{''.join(tb_str)}")
+        finally:
+            # Ensure all tasks are properly cancelled
+            for task in asyncio.all_tasks():
+                if task is not asyncio.current_task():
+                    task.cancel()
+            await asyncio.gather(*asyncio.all_tasks(), return_exceptions=True)
+
+    def _handle_signal(self, signal_number):
```

Comment:
```suggestion
    def _handle_signal(self, signal_number: int):
```

Reply:
Fixed in https://github.com/wazuh/wazuh/commit/47e4508357e0fb53f499847d6a791a0f4c86f15b.